### ISSUE TITLE
Grouch should transform data to lowercase.

### ISSUE THREAD

**Issue by casey2161** [classrank/Grouch#15]
Currently grouch uses whatever it directly pulls; however, I believe for consistency/ease of use with databases grouch should transform all data to lower case.

**Comment by mmanguno** [comment]
Rather than forcing to lowercase, we should instead provide some interface for transformations to interact with the data. That way, Grouch still retains a focus on its main functionality (scraping), and we can separate concerns a bit better.

In this case, we could define some lowercase-transformation, _L_:

``` python
L = lambda x : x.lower()
```

With such an interface, users could provide multiple avenues by which to transform and receive their data, and could still retain the original casing of OSCAR (if they so desire).

We could possibly have a standard library of transformations (or a few example ones), and make a command-line flag (`--transformer=` or something like that) for users to provide alternatives. If a user does not provide any transformation, Grouch could use a default identity transformation, _I_:

``` python
I = lambda x: x
```

**Comment by joshuamorton** [comment]
Seems reasonable, if a little extreme.

**Comment by casey2161** [comment]
I think that it solves other issues related to classrank. As well as just making the tool generally more useful.

**Comment by joshuamorton** [comment]
I agree, it just feels a tad over engineered, but sane defaults minimize that issue

**Comment by mmanguno** [comment]
For the moment, we could just pass in some flag on the command line for lower-casing the results.

**Comment by mmanguno** [comment]
Actually, doing the opposite would probably be preferable: default to lowercase, and give an option to retain the original case.
